modules/Router.js

```diff
@@ -15,5 +15,39 @@
    * @class  Router
    * @author kent.willis@vecna.com
    */
-  return Backbone.Router.extend({});
+
+  var Router = Backbone.Router.extend({
+
+    /**
+    * overridden the route function to trigger start and end signals
+    */
+    route: function(route, name, callback) {
+      var routeName = route;
+      if (!_.isRegExp(route)) route = this._routeToRegExp(route);
+      if (_.isFunction(name)) {
+        callback = name;
+        name = '';
+      }
+      if (!callback) callback = this[name];
+      var callbackCopy = function(){
+        var uuid = (new Date()).getTime().toString(16)+Math.floor(1E7*Math.random()).toString(16);
+        this.trigger('routeTiming', 
```

Comment:
See all the comments about these type of event names and payloads.

Note, the convention of triggered events are "-" separated (not camelCased) amd that they describe what has or about to happen.